toonz/sources/toonz/ocaio.cpp

```diff
@@ -534,4 +535,448 @@ void ExportOCACommand::execute() {
+    // or "RGBM" as it's called,
+    // make it very paintfull to not have blending mode while animating... 
+    // (ex : shoot paper drawings and they are opaque rasters, so onion skin doesn' work.
+    // Ok you can set white level in the capture setting image ajust,
+    // and check White As Transparent in the raster level settings,
+    // The level premult option doesn't make onion skin work, if white level is too low...
+    // but that's destructive.
+    // And the level premult option doesn't make onion skin work, if white level is too low...
+    // I would even add a realtime shader for chromakey and lumakey in the column header 
+    // (even though i agree that it would be painfull to have all necessary keying parameters exposed column header in the ui)
+    // - Frame tags
+    // NavigationTags (frame tags) should be supported by OCA
+    // NavigationTags *navTags = xsh->getNavigationTags();
+    // this could be a great base to implement an onion skin filter... (like TVPaint has)
+    // (ex : onion skin displays only 1 prev + 1 next key drawing, etc...)
+    // Unfortunately, OCA doesn't support that.
+
+    // recursion for sublayers : nothing special to do ?
+    // see CHILD_XSHLEVEL ?
+    for (auto layer : jsonLayer["childLayers"].toArray()) {
+      // I guess we dont need to deal with jsonLayer["passThrough"] ?
+      // when passThrough is false, the group content *must* be merged in rendering, 
+      // else the group is only used as a way to group the layers in the UI of the application
+      importOcaLayer(layer.toObject());
+    }
+
+    TApp::instance()->getCurrentLevel()->setLevel(
+        level->getSimpleLevel());  // selects the last created level
+    TApp::instance()->getCurrentLevel()->notifyLevelChange();
+    TApp::instance()->getCurrentScene()->notifyCastChange();
+
+  } else {
+    DVGui::warning(QObject::tr("importOcaLayer skipping unimplemented layer type : ") +
+        jsonLayer["type"].toString() + " : " + jsonLayer["name"].toString()
+    );
+  }
+}
+
+/// <summary>
+/// creates frame from OCA frame
+/// see OCA frame specs
+/// https://github.com/RxLaboratory/OCA/blob/master/src-docs/docs/specs/frame.md
+/// UNUSED properties :
+/// - meta: metadata...
+/// - opacity: per frame opacity isn't used.
+/// - position: transforms handled in a separate file ? (keyframe animation transfert for camera, pegs, etc...) 
+/// And what about scale/rotation ?
+/// - name: could be used to create empty frames (named "_blank")
+/// </summary>
+/// <param name="jsonLayer"></param>
+
+void OCAIo::OCAInputData::importOcaFrame(const QJsonObject &jsonFrame,
+                                         TXshSimpleLevel *sl) {
+  // doesExists (QFileInfo.exists()) should skip empty frames ! but...
+  TFrameId fid(jsonFrame["frameNumber"].toInt());
+  if (jsonFrame["name"].toString() == "_blank" ||
+      jsonFrame["fileName"].toString() == "") {
+    TImageP img = sl->createEmptyFrame();
+    sl->setFrame(fid, img);
```

Comment:
I think this is incorrect.  We should be skipping "_blank" frames completely not creating empty image frames.

Based on what I see generated from the OCR Export, the "_blank" frames are actually "empty" cells with no image.  Here you are creating a frame with an empty image.

Blank frames, in T2D are represented by a specific FID (-1, I think) without a level.  I am not at my computer to check, but "blank" frames are not added to the level so we should not be creating these.